src/main/kotlin/org/rust/ide/intentions/createFromUsage/CreateStructIntention.kt

```diff
@@ -50,9 +48,23 @@ class CreateStructIntention : RsElementBaseIntentionAction<CreateStructIntention
 
         val fields = inserted.blockFields?.namedFieldDeclList.orEmpty()
         if (inserted.containingFile == function.containingFile && fields.isNotEmpty()) {
-            editor.caretModel.moveToOffset(fields[0].textOffset)
-            RsMultipleVariableRenamer(fields.map { it.createSmartPointer() })
-                .doRename(fields[0], editor, DataManager.getInstance().getDataContext(editor.component))
+            val unknownTypes = inserted.descendantsOfType<RsBaseType>().filter { it.underscore != null }
```

Comment:
Let's map them to smart pointers, and `fields` as well. Raw PSI may not outlive `newTemplateBuilder` call.

Reply:
Changed to smart pointers. I was on a vacation, thus the delay.